pkg/controllers/status/controller.go

```diff
@@ -495,24 +457,27 @@ func (s *StatusController) clusterNames() ([]string, error) {
 // clusterStatuses returns the resource status in member cluster.
 func (s *StatusController) clusterStatuses(
 	ctx context.Context,
-	fedObject *unstructured.Unstructured,
+	fedObject fedcorev1a1.GenericFederatedObject,
+	targetQualifiedName common.QualifiedName,
+	targetGVK schema.GroupVersionKind,
+	typeConfig *fedcorev1a1.FederatedTypeConfig,
 	clusterNames []string,
-	qualifiedName common.QualifiedName,
-) []util.ResourceClusterStatus {
-	clusterStatus := []util.ResourceClusterStatus{}
+) []fedcorev1a1.CollectedFieldsWithCluster {
+	clusterStatus := []fedcorev1a1.CollectedFieldsWithCluster{}
 	keyedLogger := klog.FromContext(ctx)
 	// collect errors during status collection and record them as event
-	errList := []string{}
+	var errList []string
 
 	for _, clusterName := range clusterNames {
-		resourceClusterStatus := util.ResourceClusterStatus{ClusterName: clusterName}
+		resourceClusterStatus := fedcorev1a1.CollectedFieldsWithCluster{Cluster: clusterName}
 
-		clusterObj, exist, err := util.GetClusterObject(
+		clusterObj, exist, err := informermanager.GetClusterObject(
```

Comment:
How about reenqueuing if the informer is not yet synced instead? If we create a new FTC,  the informer for the source type won't be synced for a while + all objects of the source type will be enqueued at the same time. This would lead to a burst of requests to apiserver, potentially affecting the apiserver or causing ourselves to be rate-llimited.
cc @gary-lgy

Reply:
We can explore this issue in a new PR instead. This will do for now.